sql/moz-fx-data-marketing-prod/google_search_console_derived/search_impressions_by_site_v2/metadata.yaml

```diff
@@ -21,9 +21,22 @@ scheduling:
   date_partition_parameter: date
   # Google Search Console exports typically happen two days after the data date in UTC.
   date_partition_offset: -1
-  depends_on:
-  - task_id: google_search_console_derived__site_impression_exports_empty_check__v1
-    dag_name: bqetl_google_search_console
+  depends_on_table_partitions_existing:
```

Comment:
It could be, but personally I don't see a big benefit to doing so, and it seems like it'd require some advanced `cattrs` code for polymorphism. (or maybe you meant to have the sub-items as keys rather than a list? but even then I'm not sure it'd be worth it)